c11threads.h

```diff
@@ -194,90 +436,153 @@ static inline int mtx_timedlock(mtx_t *mtx, const struct timespec *ts)
+	return _cnd_init_win32(cond);
+#else
 	return pthread_cond_init(cond, 0) == 0 ? thrd_success : thrd_error;
+#endif
 }
 
 static inline void cnd_destroy(cnd_t *cond)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	(void)cond;
+#else
 	pthread_cond_destroy(cond);
+#endif
 }
 
 static inline int cnd_signal(cnd_t *cond)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	return _cnd_signal_win32(cond);
+#else
 	return pthread_cond_signal(cond) == 0 ? thrd_success : thrd_error;
+#endif
 }
 
 static inline int cnd_broadcast(cnd_t *cond)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	return _cnd_broadcast_win32(cond);
+#else
 	return pthread_cond_broadcast(cond) == 0 ? thrd_success : thrd_error;
+#endif
 }
 
 static inline int cnd_wait(cnd_t *cond, mtx_t *mtx)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	return _cnd_wait_win32(cond, mtx);
+#else
 	return pthread_cond_wait(cond, mtx) == 0 ? thrd_success : thrd_error;
+#endif
 }
 
 static inline int cnd_timedwait(cnd_t *cond, mtx_t *mtx, const struct timespec *ts)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	_cnd_timedwait_win32(cond, mtx, ts);
+#else
 	int res;
 
 	if((res = pthread_cond_timedwait(cond, mtx, ts)) != 0) {
 		return res == ETIMEDOUT ? thrd_timedout : thrd_error;
 	}
 	return thrd_success;
+#endif
 }
+#endif
 
 /* ---- thread-specific data ---- */
 
 static inline int tss_create(tss_t *key, tss_dtor_t dtor)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	return _tss_create_win32(key, dtor);
+#else
 	return pthread_key_create(key, dtor) == 0 ? thrd_success : thrd_error;
+#endif
 }
 
 static inline void tss_delete(tss_t key)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	_tss_delete_win32(key);
+#else
 	pthread_key_delete(key);
+#endif
 }
 
 static inline int tss_set(tss_t key, void *val)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	return _tss_set_win32(key, val);
+#else
 	return pthread_setspecific(key, val) == 0 ? thrd_success : thrd_error;
+#endif
 }
 
 static inline void *tss_get(tss_t key)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	return _tss_get_win32(key);
+#else
 	return pthread_getspecific(key);
+#endif
 }
 
 /* ---- misc ---- */
 
+#if !defined(_WIN32) || defined(C11THREADS_PTHREAD_WIN32) || !defined(C11THREADS_SUPPORT_WINNT_OLDER_THAN_VISTA)
 static inline void call_once(once_flag *flag, void (*func)(void))
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	_call_once_win32(flag, func);
+#else
 	pthread_once(flag, func);
+#endif
 }
+#endif
 
-#if defined(C11THREADS_NO_TIMESPEC_GET)
-/* TODO take base into account */
+#ifdef C11THREADS_NO_TIMESPEC_GET
 static inline int timespec_get(struct timespec *ts, int base)
 {
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	return _timespec_get_win32(ts, base);
+#else
 	struct timeval tv;
 
-	gettimeofday(&tv, 0);
+	if(base != TIME_UTC) {
+		return 0;
+	}
```

Comment:
TIME_UTC is only available in C11. I'd rather have this code be able to build with C99 or even C89 compilers if at all possible. Is this check necessary? What's the purpose of it?

Reply:
Clicked the wrong button... TL;DR of what I just wrote:

At least one base is needed. The function is supposed to return 0 on failure and `base` on success. Microsoft, for example, expects `TIME_UTC` and their function fails if anything else is provided as base.

Further documentation: https://en.cppreference.com/w/c/chrono/timespec_get

Reply:
Forgot to mention: c11threads.h will define `TIME_UTC` if it's not already defined.

Reply:
That's perfectly fine then. Because previously it broke the build here, with TIME_UTC undefined, while compiling with `-std=gnu99` (C99 + extensions).